### DISCUSSION TITLE
Understanding framelinvel

### DISCUSSION THREAD

**Discussion by nilsiism** [google-deepmind/mujoco#1934] [Asking for Help]
Hello Community,

I have a problem understanding the `framelinvel` sensor

I added three sensors

```xml
<framelinvel name="fooVelBase" objtype="site" objname="Foo" reftype="body" refname="Bar"/>
<framelinvel name="fooVelWorld" objtype="site" objname="Foo"/>
<framelinvel name="barVelWorld" objtype="body" objname="Bar"/>
```

Next I read their sensor values and take the norm

```python
bar_lin_vel_W = mjdata.sensor("barVelWorld").data
norm_bar_lin_vel_W = np.linalg.norm(bar_lin_vel_W)

foo_lin_vel_W = mjdata.sensor("fooVelWorld").data
norm_foo_lin_vel_W = np.linalg.norm(foo_lin_vel_W)

foo_lin_vel_B = mjdata.sensor("fooVelBase").data
norm_foo_lin_vel_B = np.linalg.norm(foo_lin_vel_B)
```

Due to the triangle inequality I would have assume that the velocity of `bar` is bound by

```python
norm_foo_lin_vel_W - norm_foo_lin_vel_B <= norm_bar_lin_vel_W <= norm_foo_lin_vel_W + norm_foo_lin_vel_B
```

However I get the following values
```python
norm_bar_lin_vel_W = 0.576
norm_foo_lin_vel_W = 1.290
norm_foo_lin_vel_B =  0.249
```

Could someone help me where I'm misunderstanding the velocities?
Any help is much appreciated

**Comment by yuvaltassa** [comment]
Your assumption is incorrect. Consider the following scenario
1. Body `b` is at the origin, spinning around the z axis at 1 rad/sec, no linear velocity.
2. Site `s` is at (10, 0, 0), fixed to the world.
3. The linear velocity of both `b` and `s` is 0 in the world frame. However in the (rotating) reference frame of the body, the fixed site appears to be moving with a linear velocity of 10 m/s.

Makes sense?
